source/functions.cpp

```diff
@@ -46,146 +51,69 @@ void print(HCL::variable msg, std::string end/* = \n*/) {
 
 	if (msg.value.size() > 1) output += "}";
 
-	printf("%s%s", output.c_str(), end.c_str());
+	std::printf("%s%s", output.c_str(), end.c_str());
 }
 
 
-int createFolder(std::string path, int mode/* = 0777*/) {
-	int check;
-	std::string fullPath;
-	std::vector<std::string> folders = split(path, "/");
-
-	for (auto folder : folders) {
-		fullPath += folder;
-		check = mkdir(fullPath.c_str(), mode);
-		fullPath += "/";
-	}
-  
-    return check;
+int createFolder(std::string path) {
```

Comment:
It should be noted that at [`core.cpp` from line 285 to 300](https://github.com/Frityet/HCL/blob/main/source/core.cpp#L285-L300), code should be changed to accommodate the parameter changes (since we only have to check 1 parameter and no more default arguments). This is how that would look:
```c++
	// int createFolder(string path)
	else if (useFunction("int", "createFolder", 1, 1)) {
		if (params[0].type != "string") HCL::throwError(true, "Cannot input a '%s' type to a string-only parameter (param 'path' is string-only)", true, params[0].type.c_str());


		int result = createFolder(params[0].value[0]);
		return intToVoid(result);  // Returns the result as int.
	}
```

Reply:
Along side this, using `void *` in C++ is a really bad idea, and casting an int to a `void *` is a really really bad idea, maybe this should be changed?

Reply:
Using `void*` can be bad if it isn't used carefully. However, in practice the type hasn't caused any issues on my part as I only cast to it and convert it back to the original type as I have the privilege of knowing the correct type each time I need a function's output and as such, any related memory bug issues are unlikely to happen in terms of using `void*`. Besides, what other alternatives would there be for a generic variable that can be an int, string and what not?